testsuite/certificates/__init__.py

```diff
@@ -30,6 +33,31 @@ class Certificate:
     certificate: str
     chain: str
 
+    @cached_property
+    def decoded(self) -> x509.Certificate:
```

Comment:
since the Certificate dataclass is not frozen, these cached_properties may become outdated and invalid when underlying certificate property is changed